sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala

```diff
@@ -183,6 +186,57 @@ class CollationSuite extends DatasourceV2SQLBase {
     }
   }
 
+  test("aggregates count respects collation") {
```

Comment:
maybe we should also test struct columns with collations inside

Reply:
Added test case to golden file. Please take a look.